crates/red_knot_test/src/assertion.rs

```diff
@@ -0,0 +1,619 @@
+//! Parse type and type-error assertions in Python comment form.
```

Comment:
I'm wondering if some or all of this doc-comment should actually be a `README.md` for the crate as a whole. It would be really useful to point contributors to some comprehensive docs on how we write our tests. We could always port an expanded version of this doc-comment to a README as a followup PR, though; this comment of mine isn't blocking.

Reply:
Good point. I mentioned that in my previous review but forgot to check if it's there. I strongly agree that we should document the test format.

Reply:
Yeah, I forgot to mention this; I actually already started on a `README.md` for the test framework but it's not done yet. At this point I think I will finish it tomorrow and submit it as a separate PR. Definitely agree this is important.